loki/batch/scheduler.py

```diff
@@ -377,6 +377,46 @@ def rekey_item_cache(self):
         )
 
     def process(self, transformation):
+        """
+        Process all :attr:`items` in the scheduler's graph with either
+        a :any:`Pipeline` or a single :any:`Transformation`.
+
+        A single :any:`Transformation` pass invokes
+        :meth:`process_transformation` individually, while a
+        :any:`Pipeline` will apply each contrained transformation in
+        turn over the full dependency graph of the scheduler.
+
+        Parameters
+        ----------
+        transformation : :any:`Transformation` or :any:`Pipeline`
+            The transformation or transformation pipeline to apply
+        """
+        from loki.transform import Transformation, Pipeline  # pylint: disable=import-outside-toplevel
+
+        if isinstance(transformation, Transformation):
+            self.process_transformation(transformation=transformation)
+
+        elif isinstance(transformation, Pipeline):
+            self.process_pipeline(pipeline=transformation)
+
+        else:
+            error('[Loki::Scheduler] Batch processing requires Transformation or Pipeline object')
+            raise RuntimeError('[Loki] Could not batch process {transformation_or_pipeline}')
+
+    def process_pipeline(self, pipeline):
+        """
+        Process a given :any:`Pipeline` by applying its assocaited
+        transformations in turn.
+
+        Parameters
+        ----------
+        transformation : :any:`Pipeline`
+            The transformation pipeline to apply
+        """
+        for transformation in pipeline.transformations:
+            self.process_transformation(transformation)
```

Comment:
No, in the follow-up PR #271 we introduce native addition of pipelines and transformations, and they always result in a new pipeline, eg. `pipeline = short_pipeline + other_pipeline + some_trafo` will result in a single `Pipeline` object that encapsulates a single sequence of `Transformation` objects by appending them as the notation suggests. 

A "pipeline of pipelines" could potentially become problematic if we extend the abstraction with further mechanics, so I'd like to avoid that. We could maybe make "list of Pipeline" a thing, but I feel that's beyond the intro PR. :wink: